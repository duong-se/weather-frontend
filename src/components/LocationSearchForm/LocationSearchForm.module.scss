.searchButton {
  border: 1px solid transparent;
  padding: 0.55rem 0.75rem;
}
.searchIcon {
  display: flex;
  align-items: center;
  margin-bottom: 0;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  text-align: center;
  white-space: nowrap;
  border-radius: 0;
}