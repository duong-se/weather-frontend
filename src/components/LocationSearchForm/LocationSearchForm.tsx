import React from 'react'
import LocationInput from '../LocationInput'
import Spin from '../Spin'

export type LocationSearchFormProps = {
  onSubmit: (e: React.BaseSyntheticEvent<any>) => void
  isLoading: boolean
}

export const LocationSearchForm: React.FC<LocationSearchFormProps> = ({ onSubmit, isLoading }) => {
  return (
    <form className="w-100" onSubmit={onSubmit}>
      <div className="row">
        <div className="col-sm-8 col-md-8 col-lg-3 mb-4">
          <LocationInput />
        </div>
        <div className="col-sm-4 cold-md-4 col-lg-3 mb-4">
          <button type="submit" className="btn btn-primary w-100">
            {isLoading ? <Spin /> : 'Submit'}
          </button>
        </div>
      </div>
    </form>
  )
}

export default LocationSearchForm
