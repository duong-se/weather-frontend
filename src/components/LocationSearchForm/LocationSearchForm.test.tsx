import React from 'react'
import { shallow } from 'enzyme'
import { LocationSearchForm } from './LocationSearchForm'

describe('LocationSearchForm', () => {
  it('should match snapshot', () => {
    const onSubmit = jest.fn()
    const isLoading = true
    const wrapper = shallow(<LocationSearchForm onSubmit={onSubmit} isLoading={isLoading} />)
    const submitBtn = wrapper.find('.btn').text()
    expect(submitBtn).not.toEqual('Submit')
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot', () => {
    const onSubmit = jest.fn()
    const isLoading = false
    const wrapper = shallow(<LocationSearchForm onSubmit={onSubmit} isLoading={isLoading} />)
    const submitBtn = wrapper.find('.btn').text()
    expect(submitBtn).toEqual('Submit')
    expect(wrapper).toMatchSnapshot()
  })

  it('should call onSubmit', () => {
    const onSubmit = jest.fn()
    const isLoading = false
    const wrapper = shallow(<LocationSearchForm onSubmit={onSubmit} isLoading={isLoading} />)
    const formWrapper = wrapper.find('form')
    formWrapper.simulate('submit')
    expect(onSubmit).toBeCalled()
  })
})
