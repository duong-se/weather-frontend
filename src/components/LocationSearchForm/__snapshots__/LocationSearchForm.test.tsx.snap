// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSearchForm should match snapshot 1`] = `
<form
  className="w-100"
  onSubmit={[MockFunction]}
>
  <div
    className="row"
  >
    <div
      className="col-sm-8 col-md-8 col-lg-3 mb-4"
    >
      <TextInput />
    </div>
    <div
      className="col-sm-4 cold-md-4 col-lg-3 mb-4"
    >
      <button
        className="btn btn-primary w-100"
        type="submit"
      >
        <Spin />
      </button>
    </div>
  </div>
</form>
`;

exports[`LocationSearchForm should match snapshot 2`] = `
<form
  className="w-100"
  onSubmit={[MockFunction]}
>
  <div
    className="row"
  >
    <div
      className="col-sm-8 col-md-8 col-lg-3 mb-4"
    >
      <TextInput />
    </div>
    <div
      className="col-sm-4 cold-md-4 col-lg-3 mb-4"
    >
      <button
        className="btn btn-primary w-100"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
